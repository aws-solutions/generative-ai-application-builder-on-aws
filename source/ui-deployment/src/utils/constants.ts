/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

export const API_NAME = 'api';
export const APP_TRADEMARK_NAME = 'Generative AI Application Builder on AWS';

export const MIN_USE_CASE_NAME_LENGTH = 1;
export const MAX_USE_CASE_NAME_LENGTH = 64;
export const MIN_KENDRA_IDX_NAME_LENGTH = 1;
export const MAX_KENDRA_IDX_NAME_LENGTH = 64;
export const MIN_USE_CASE_DESCRIPTION_LENGTH = 0;
export const MAX_USE_CASE_DESCRIPTION_LENGTH = 200;
export const MIN_MODEL_NAME_LENGTH = 1;
export const MAX_MODEL_NAME_LENGTH = 64;
export const MIN_API_KEY_LENGTH = 1;
export const MAX_API_KEY_LENGTH = 128;
export const MIN_PROMPT_TEMPLATE_LENGTH = 0;
export const MAX_PROMPT_TEMPLATE_LENGTH = 1000;
export const MIN_MODEL_PARAMETER_NAME_LENGTH = 0;
export const MAX_MODEL_PARAMETER_NAME_LENGTH = 64;
export const MIN_MODEL_PARAMETER_VALUE_LENGTH = 0;
export const MAX_MODEL_PARAMETER_VALUE_LENGTH = 2000;
export const MIN_KENDRA_INDEX_ID_LENGTH = 0;
export const MAX_KENDRA_INDEX_ID_LENGTH = 64;
export const MIN_ADDITIONAL_KENDRA_QUERY_CAPACITY = 0;
export const MAX_ADDITIONAL_KENDRA_QUERY_CAPACITY = 5;
export const MIN_ADDITIONAL_KENDRA_STORAGE_CAPACITY = 0;
export const MAX_ADDITIONAL_KENDRA_STORAGE_CAPACITY = 5;
export const MIN_KNOWLEDGE_BASE_NUM_DOCS = 1;
export const MAX_KNOWLEDGE_BASE_NUM_DOCS = 5;
export const DELAY_AFTER_DELETE_MS = 2000; // 2 seconds

export const DEFAULT_KENDRA_NUMBER_OF_DOCS = 2;
export const DEFAULT_ADDITIONAL_KENDRA_QUERY_CAPACITY = 0;
export const DEFAULT_ADDITIONAL_KENDRA_STORAGE_CAPACITY = 0;

export const DEPLOYMENT_PLATFORM_API_ROUTES = {
    LIST_USE_CASES: { route: '/deployments', method: 'GET' },
    CREATE_USE_CASE: { route: '/deployments', method: 'POST' },
    UPDATE_USE_CASE: {
        route: (useCaseId: string) => {
            return `/deployments/${useCaseId}`;
        },
        method: 'PATCH'
    },
    DELETE_USE_CASE: {
        route: (useCaseId: string) => {
            return `/deployments/${useCaseId}`;
        },
        method: 'DELETE'
    }
};

export const DEPLOYMENT_STATUS_NOTIFICATION = {
    SUCCESS: 'SUCCESS',
    FAILURE: 'FAILURE',
    PENDING: 'PENDING'
};

export const DEPLOYMENT_ACTIONS = {
    CREATE: 'CREATE',
    EDIT: 'EDIT',
    CLONE: 'CLONE'
};

export const deploymentActionText = {
    [DEPLOYMENT_ACTIONS.CREATE]: 'Deploy',
    [DEPLOYMENT_ACTIONS.EDIT]: 'Update',
    [DEPLOYMENT_ACTIONS.CLONE]: 'Clone'
};

export const CFN_STACK_STATUS_INDICATOR = {
    SUCCESS: 'success',
    ERROR: 'error',
    IN_PROGRESS: 'in-progress',
    WARNING: 'warning',
    STOPPED: 'stopped'
};

export const MODEL_PARAM_TYPES = ['string', 'integer', 'float', 'boolean', 'list', 'dict'];

export const DEFAULT_KNOWLEDGE_BASE_TYPE = 'Kendra';

export const LEGAL_DISCLAIMER = `
Generative AI Application Builder on AWS allows you to build and deploy
generative artificial intelligence (GAI) applications on AWS by engaging the GAI model
of your choice, including third-party GAI models that you may choose to use that AWS
does not own or otherwise have any control over (“Third-Party GAI Models”).
Your use of the Third-Party GAI Models is governed by the terms provided to you
by the Third-Party GAI Model providers when you acquired your license to use them
(for example, their terms of service, license agreement, acceptable use policy, and privacy policy).
You are responsible for ensuring that your use of the Third-Party GAI Models comply
with the terms governing them, and any laws, rules, regulations, policies, or standards
that apply to you. You are also responsible for making your own independent assessment
of the Third-Party GAI Models that you use, including their outputs and how Third-Party
GAI Model providers use any data that may be transmitted to them based on your deployment
configuration. AWS does not make any representations, warranties, or guarantees regarding
the Third-Party GAI Models, which are “Third-Party Content” under your agreement with AWS.
Generative AI Application Builder on AWS is offered to you as
“AWS Content” under your agreement with AWS.
`;

export const KENDRA_WARNING = `
Amazon Kendra index created here is NOT deleted automatically when the use-case is deleted. You are responsible for deleting the index when you no longer require it. Kendra charges apply while the index is active.`;

export const INTERNAL_USER_GENAI_POLICY_URL = 'https://policy.a2z.com/docs/568686/publication';

export const SUPPORTED_FALCON_MODELS = [
    'tiiuae/falcon-40b-instruct',
    'tiiuae/falcon-7b-instruct',
    'tiiuae/falcon-40b',
    'tiiuae/falcon-7b',
    'google/flan-t5-xxl',
    'google/flan-t5-xl',
    'google/flan-t5-large',
    'google/flan-t5-base',
    'google/flan-t5-small'
];

export const BEDROCK_MODEL_PROVIDER_NAME = 'Bedrock';
export const ERROR_MESSAGES = {
    UNAUTHORIZED: 'Request failed with status code 403'
};

const IG_ROOT = 'https://docs.aws.amazon.com/solutions/latest/generative-ai-application-builder-on-aws'
export const IG_DOCS = {
    BEDROCK_SECURITY: `${IG_ROOT}/security-1.html#using-third-party-models-on-amazon-bedrock`,
    CHOOSING_LLMS: `${IG_ROOT}/use-the-solution.html#choosing-the-right-llm-for-your-use-case`,
    CONCEPTS: `${IG_ROOT}/concepts-and-definitions.html`,
    COST: `${IG_ROOT}/cost.html`,
    INGESTING_DATA: `${IG_ROOT}/step-3-ingest-data-into-knowledge-base.html`,
    MANAGE_USERS: `${IG_ROOT}/customization-guide.html#managing-cognito-user-pool`,
    SUPPORTED_LLMS: `${IG_ROOT}/supplemental-topics.html#supported-llm-providers`,
    THIRD_PARTY_SECURITY: `${IG_ROOT}/security-1.html#third-party-llm-integrations-outside-of-amazon-bedrock`,
    TIPS_PROMPT_LIMITS: `${IG_ROOT}/use-the-solution.html#tips-for-managing-model-token-limits`,
    UNINSTALL: `${IG_ROOT}/uninstall-the-solution.html`,
    USE_CASES: `${IG_ROOT}/use-cases.html`,
    USING_THE_SOLUTION: `${IG_ROOT}/use-the-solution.html`,
}

export const LANDING_PAGE_URL = 'https://aws.amazon.com/solutions/implementations/generative-ai-application-builder-on-aws/'