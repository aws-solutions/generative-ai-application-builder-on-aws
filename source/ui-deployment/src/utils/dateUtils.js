// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export const dateOptions = {
    weekday: 'short',
    year: 'numeric',
    month: 'short',
    day: 'numeric',
    hour: 'numeric',
    minute: 'numeric'
};