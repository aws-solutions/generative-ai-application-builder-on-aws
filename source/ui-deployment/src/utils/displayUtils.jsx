// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { Box } from '@cloudscape-design/components';

export const escapedNewLineToLineBreakTag = (str, componentId) => {
    try {
        return str.split('\n').map((item, index) => {
            return index === 0 ? item : [<br key={`${componentId}.${index}`} />, item]; //NOSONAR - split new line characters to html br tags once
        });
    } catch (error) {
        return '';
    }
};

export const createBox = (data) => {
    return <Box variant="p">{data}</Box>;
};