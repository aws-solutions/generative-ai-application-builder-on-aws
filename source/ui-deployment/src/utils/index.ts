// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export * from './test-utils';
export * from './utils';
export * from './dateUtils';
export * from './displayUtils';
export * from './linkUtils';
export * from './tools-content';
export * from './ValueWithLabel';
