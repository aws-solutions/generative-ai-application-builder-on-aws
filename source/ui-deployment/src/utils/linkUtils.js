// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

// function to create a direct link to the Kendra console given a Kendra index id
export const createKendraConsoleLink = (region, kendraIndexId) => {
    return `https://console.aws.amazon.com/kendra/home?region=${region}#/indexes/${kendraIndexId}/details`;
};

// Note: there currently does not exist a direct link given a knowledge base ID
// the console link uses the KB name in the URL, which we do not have access to at this point
export const createBedrockKnowledgeBaseConsoleLink = (region, bedrockKnowledgeBaseId) => {
    return `https://console.aws.amazon.com/bedrock/home?region=${region}#/knowledge-bases`;
};

export const createVpcLink = (region, vpcId) => {
    return `https://console.aws.amazon.com/vpcconsole/home?region=${region}#VpcDetails:VpcId=${vpcId}`;
};