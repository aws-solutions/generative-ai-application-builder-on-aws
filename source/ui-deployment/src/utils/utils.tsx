// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { Auth } from 'aws-amplify';

export interface TransformedDeployment {
    useCaseUUID: string;
    defaultUserEmail: string;
    IsInternalUser?: string;
    createNewVpc: string;
    UseCaseId?: string;
    CreatedDate?: string;
    StackId?: string;
    status?: string;
    UseCaseType?: string;
    UseCaseName?: string;
    Name?: string;
    Description?: string;
    ragEnabled?: string;
    deployUI?: string;
    vpcEnabled?: string;
    vpcId?: string;
    cloudwatchDashboardUrl?: string;
    cloudFrontWebUrl?: string;
    knowledgeBaseType?: string;
    kendraIndexId?: string;
    bedrockKnowledgeBaseId?: string;
    privateSubnetIds?: string[];
    securityGroupIds?: string[];
    KnowledgeBaseParams?: any;
    ConversationMemoryParams?: any;
    LlmParams?: any;
    FeedbackParams?: any;
    AuthenticationParams?: any;
    AgentParams?: any;
    [key: string]: any; // Allow for additional properties
}

/**
 * Use currently authenticated user to generate a JWT token from
 * Cognito.
 * @returns JWT Token
 */
export async function generateToken() {
    try {
        const user = await Auth.currentAuthenticatedUser();
        const token = user.getSignInUserSession().getAccessToken().getJwtToken();
        return token;
    } catch (error) {
        console.error('error REST API:', error);
    }
}

// Copy properties from API response, preserving structure
export const copyIfDefined = (source: any, target: any, sourceKey: string, targetKey?: string) => {
    if (source[sourceKey] !== undefined) {
        target[targetKey || sourceKey] = source[sourceKey];
    }
};

/**
 * Transforms the API response format to match the existing selectedDeployment format
 * preserving the original structure without adding undefined attributes
 * @param apiResponse The response from the getUseCaseDetails API
 * @returns An object matching the format of the existing selectedDeployment
 */
export const mapApiResponseToSelectedDeployment = (apiResponse: any): TransformedDeployment | null => {
    if (!apiResponse) {
        return null;
    }

    // Extract the UUID from the UseCaseId
    const useCaseUUID = apiResponse.UseCaseId ? apiResponse.UseCaseId.substring(0, 8) : '';

    // Create base object with required wizard fields
    const transformedDeployment: TransformedDeployment = {
        useCaseUUID,
        defaultUserEmail: 'placeholder@example.com',
        createNewVpc: 'No'
    };

    // Copy all fields from the API response
    const fields = [
        'UseCaseId',
        'CreatedDate',
        'StackId',
        'UseCaseType',
        'Description',
        'ragEnabled',
        'deployUI',
        'vpcEnabled',
        'vpcId',
        'cloudwatchDashboardUrl',
        'cloudFrontWebUrl',
        'knowledgeBaseType',
        'kendraIndexId',
        'bedrockKnowledgeBaseId',
        'privateSubnetIds',
        'securityGroupIds',
        'KnowledgeBaseParams',
        'ConversationMemoryParams',
        'LlmParams',
        'FeedbackParams',
        'AuthenticationParams',
        'AgentParams'
    ];

    fields.forEach((field) => copyIfDefined(apiResponse, transformedDeployment, field));

    // Special case for Status -> status
    copyIfDefined(apiResponse, transformedDeployment, 'Status', 'status');

    // Special case for UseCaseName -> both UseCaseName and Name
    if (apiResponse.UseCaseName !== undefined) {
        transformedDeployment.UseCaseName = apiResponse.UseCaseName;
        transformedDeployment.Name = apiResponse.UseCaseName;
    }

    // Special case for Bedrock knowledge base ID from nested structure
    if (
        apiResponse.bedrockKnowledgeBaseId === undefined &&
        apiResponse.KnowledgeBaseParams?.BedrockKnowledgeBaseParams?.BedrockKnowledgeBaseId !== undefined
    ) {
        transformedDeployment.bedrockKnowledgeBaseId =
            apiResponse.KnowledgeBaseParams.BedrockKnowledgeBaseParams.BedrockKnowledgeBaseId;
    }

    return transformedDeployment;
};
