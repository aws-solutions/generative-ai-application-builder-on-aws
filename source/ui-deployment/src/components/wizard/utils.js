/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { DEFAULT_KENDRA_NUMBER_OF_DOCS, DEFAULT_KNOWLEDGE_BASE_TYPE } from '../../utils/constants';
import {
    BEDROCK_MODEL_OPTION_IDX,
    HF_INF_ENDPOINT_OPTION_IDX,
    HF_MODEL_OPTION_IDX,
    MODEL_FAMILY_PROVIDER_OPTIONS
} from './steps-config';

export const getFieldOnChange =
    (fieldType, fieldKey, onChangeFn) =>
    ({ detail: { selectedOption, value, checked } }) => {
        let fieldValue;
        if (fieldType === 'select') {
            fieldValue = selectedOption;
        } else if (fieldType === 'toggle') {
            fieldValue = checked;
        } else {
            fieldValue = value;
        }
        onChangeFn({
            [fieldKey]: fieldValue
        });
    };

export const createDeployRequestPayload = (stepsInfo) => {
    const payload = {
        ...createKnowledgeBaseApiParams(stepsInfo.knowledgeBase),
        ...createLLMParamsApiParams(stepsInfo.model, stepsInfo.knowledgeBase.isRagRequired),
        ...createConversationMemoryApiParams(),
        ...createUseCaseInfoApiParams(stepsInfo.useCase, stepsInfo.model)
    };

    return payload;
};

export const createUseCaseInfoApiParams = (useCaseStepInfo, modelStepInfo) => {
    const params = {
        UseCaseName: useCaseStepInfo.useCaseName,
        UseCaseDescription: useCaseStepInfo.useCaseDescription,
        ...(useCaseStepInfo.defaultUserEmail &&
            useCaseStepInfo.defaultUserEmail !== '' && {
                DefaultUserEmail: useCaseStepInfo.defaultUserEmail
            }),
        KnowledgeBaseType: DEFAULT_KNOWLEDGE_BASE_TYPE
    };

    if (modelStepInfo.modelProvider !== MODEL_FAMILY_PROVIDER_OPTIONS[BEDROCK_MODEL_OPTION_IDX]) {
        params.ConsentToDataLeavingAWS = true;
    }
    return params;
};

export const stringListToSelectionOptions = (list) => {
    return list.map((item) => {
        return {
            label: item,
            value: item
        };
    });
};

/**
 * Transform the memory step to the params required by the API
 *
 * @param {Object} memoryStepInfo
 */
export const createConversationMemoryApiParams = (memoryStepInfo) => {
    return {
        ConversationMemoryType: 'DynamoDB'
    };
};

/**
 * Transform the model config step to the params required by the API
 * @param {Object} modelStepInfo
 * @returns
 */
export const createLLMParamsApiParams = (modelStepInfo, isRagEnabled = true) => {
    const formatValue = (value, type) => {
        let newValue = value;
        if (type === 'boolean') {
            newValue = newValue.toLowerCase();
        } else if (type === 'list') {
            if (newValue[0] !== '[') {
                newValue = '[' + newValue;
            }
            if (newValue[newValue.length - 1] !== ']') {
                newValue = newValue + ']';
            }
        }
        return newValue;
    };

    const modelParamsObjectCreator = (modelParameters) => {
        if (!modelParameters || modelParameters.length === 0) {
            return {};
        }

        const modelParamsObject = {};
        modelParameters.forEach((modelParam) => {
            modelParamsObject[modelParam.key] = {
                Value: formatValue(modelParam.value, modelParam.type.value),
                Type: modelParam.type.value
            };
        });

        return modelParamsObject;
    };

    const llmParamsPayload = {
        Streaming: modelStepInfo.streaming,
        PromptTemplate: modelStepInfo.promptTemplate,
        Verbose: modelStepInfo.verbose,
        ModelParams: modelParamsObjectCreator(modelStepInfo.modelParameters),
        Temperature: parseFloat(modelStepInfo.temperature),
        RAGEnabled: isRagEnabled
    };

    if (modelStepInfo.modelProvider !== MODEL_FAMILY_PROVIDER_OPTIONS[BEDROCK_MODEL_OPTION_IDX]) {
        llmParamsPayload.ApiKey = modelStepInfo.apiKey;
    }

    if (modelStepInfo.modelProvider === MODEL_FAMILY_PROVIDER_OPTIONS[HF_INF_ENDPOINT_OPTION_IDX]) {
        llmParamsPayload.ModelProvider = MODEL_FAMILY_PROVIDER_OPTIONS[HF_MODEL_OPTION_IDX].value;
        llmParamsPayload.InferenceEndpoint = modelStepInfo.inferenceEndpoint;
    } else {
        llmParamsPayload.ModelProvider = modelStepInfo.modelProvider.value;
        llmParamsPayload.ModelId = modelStepInfo.modelName;
    }

    const payload = {
        LlmParams: llmParamsPayload
    };
    return payload;
};

/**
 * Transform the knowledge base step to the params required by the API. The parameters
 * will be different based on whether the knowledge base step is using an existing index or not.
 * If RAG requirement is set to false in the knowledgeBase step then an empty object is returned
 * @param {Object} knowledgeBaseStep KnowledgeBase step form details
 */
export const createKnowledgeBaseApiParams = (knowledgeBaseStepInfo) => {
    if (!knowledgeBaseStepInfo.isRagRequired) {
        return {};
    }

    if (knowledgeBaseStepInfo.existingKendraIndex === 'yes') {
        // payload for using existing idx
        return {
            KnowledgeBaseParams: {
                ExistingKendraIndexId: knowledgeBaseStepInfo.kendraIndexId,
                NumberOfDocs: knowledgeBaseStepInfo.maxNumDocs ? parseInt(knowledgeBaseStepInfo.maxNumDocs) : 10,
                ReturnSourceDocs: false
            }
        };
    }

    return {
        KnowledgeBaseParams: {
            KendraIndexName: knowledgeBaseStepInfo.kendraIndexName,
            QueryCapacityUnits: knowledgeBaseStepInfo.kendraAdditionalQueryCapacity
                ? parseInt(knowledgeBaseStepInfo.kendraAdditionalQueryCapacity)
                : 0,
            StorageCapacityUnits: knowledgeBaseStepInfo.kendraAdditionalStorageCapacity
                ? parseInt(knowledgeBaseStepInfo.kendraAdditionalStorageCapacity)
                : 0,
            KendraIndexEdition: `${knowledgeBaseStepInfo.kendraEdition.value.toUpperCase()}_EDITION`,
            NumberOfDocs: knowledgeBaseStepInfo.maxNumDocs
                ? parseInt(knowledgeBaseStepInfo.maxNumDocs)
                : DEFAULT_KENDRA_NUMBER_OF_DOCS,
            ReturnSourceDocs: false
        }
    };
};

/**
 * Utility function to update the number of fields in error in entire form based on
 * the error message for a given form field.
 * @param {string} currErrorMessage
 * @param {string} errorState
 * @param {()=>void} setNumFieldsInError
 */
export const updateNumFieldsInError = (currErrorMessage, errorState, setNumFieldsInError) => {
    if (currErrorMessage.length > 0 && errorState.length === 0) {
        setNumFieldsInError((numFieldsInError) => numFieldsInError + 1);
    } else if (currErrorMessage.length === 0 && errorState.length > 0) {
        setNumFieldsInError((numFieldsInError) => numFieldsInError - 1);
    }
};
