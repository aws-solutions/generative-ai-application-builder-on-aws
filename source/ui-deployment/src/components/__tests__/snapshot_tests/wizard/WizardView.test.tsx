// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import '@testing-library/jest-dom';
import { mockReactMarkdown, snapshotWithProvider } from '@/utils';
import { TextUseCaseType } from '@/components/wizard/interfaces/UseCaseTypes/Text';
import { USECASE_TYPE_ROUTE } from '@/utils/constants';

vi.mock('@cloudscape-design/components');

let WizardView: any;
mockReactMarkdown();
WizardView = (await import('../../../wizard/WizardView')).default;

test('Snapshot test', async () => {
    const tree = snapshotWithProvider(<WizardView useCase={new TextUseCaseType()} />, USECASE_TYPE_ROUTE.TEXT).toJSON();
    expect(tree).toMatchSnapshot();
});
