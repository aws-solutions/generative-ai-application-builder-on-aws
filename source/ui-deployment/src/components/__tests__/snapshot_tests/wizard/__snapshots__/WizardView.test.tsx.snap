// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `
<div
  style={
    {
      "display": "flex",
      "justifyContent": "center",
      "padding": "20px",
    }
  }
/>
`;
